File: src/main/java/com/robohorse/robopojogenerator/generator/RoboPOJOGenerator.java
package com.robohorse.robopojogenerator.generator;

import com.robohorse.robopojogenerator.generator.common.ClassItem;
import com.robohorse.robopojogenerator.generator.processing.ClassProcessor;
import com.robohorse.robopojogenerator.models.GenerationModel;
import org.json.JSONObject;

import javax.inject.Inject;
import java.util.HashMap;
import java.util.HashSet;
import java.util.Map;
import java.util.Set;


/**
 * Created by vadim on 22.09.16.
 */
public class RoboPOJOGenerator {
    @Inject
    ClassProcessor processor;

    @Inject
    public RoboPOJOGenerator() {
    }

    public Set<ClassItem> generate(GenerationModel model) {
        final JSONObject jsonObject = new JSONObject(model.getContent());
        final Map<String, ClassItem> map = new HashMap<String, ClassItem>();
        processor.proceed(jsonObject, model.getRootClassName(), map);
        return new HashSet<ClassItem>(map.values());
    }
}

File: src/main/kotlin/com/robohorse/robopojogenerator/generator/RoboPOJOGenerator.kt
package com.robohorse.robopojogenerator.generator

import com.robohorse.robopojogenerator.generator.consts.common.ClassItem
import com.robohorse.robopojogenerator.generator.processing.ClassProcessor
import com.robohorse.robopojogenerator.generator.utils.ProcessingModelManager
import com.robohorse.robopojogenerator.models.GenerationModel
import java.util.*

class RoboPOJOGenerator(
        private val processor: ClassProcessor,
        private val processingModelManager: ProcessingModelManager
) {

    fun generate(model: GenerationModel): Set<ClassItem> {
        val map = HashMap<String, ClassItem>()
        processor.proceed(
                processingModelManager.resolveJsonModel(model),
                map
        )
        return HashSet(map.values)
    }
}
